src/main/java/com/simibubi/create/content/trains/bogey/AbstractBogeyBlock.java

```diff
@@ -227,15 +227,21 @@ public final InteractionResult use(BlockState state, Level level, BlockPos pos,
 			sbbe.setBogeyData(sbbe.getBogeyData().merge(defaultData));
 
 			if (size == getSize()) {
+				if (state.getBlock() != style.getBlockOfSize(size)) {
+					CompoundTag oldData = sbbe.getBogeyData();
+					level.setBlock(pos, copyProperties(state, getStateOfSize(sbbe, size)), 3);
```

Comment:
Please don't use bitflags like this. 
There are named constants at the top the the `Block` class that should replace the literal `3`